app/controllers/structures_controller.rb

```diff
@@ -9,14 +9,14 @@ def show
     respond_to do |format|
       format.csv do
         # Download the structural spreadsheet
-        cocina = Repository.find(params[:item_id])
+        cocina = fetch_cocina(item_id: params[:item_id])
         authorize! :update, cocina
         filename = "structure-#{Druid.new(cocina).without_namespace}.csv"
         send_data StructureSerializer.as_csv(cocina.externalIdentifier, cocina.structural), filename:
       end
       format.html do
         # Lazy loading of the structural part of the show page
-        @cocina_item = Repository.find(decrypted_token.fetch(:druid))
+        @cocina_item = fetch_cocina(item_id: decrypted_token.fetch(:druid), user_version_id: decrypted_token&.fetch(:user_version_id, nil))
```

Comment:
Do you think we still need the safe navigation in `decrypted_token&.fetch`?

Reply:
Oh no, probably not, copy-pasta.